433 Feeds: PrivateChannels and returning channels from intents (#508)
finos/FDC3: docs/api/ref/DesktopAgent.md

fdc3.addIntentListener("CreateOrder", (context) => {
  return new Promise<Context>((resolve) => {
    // go create the order
    resolve({type: "fdc3.order", id: { "orderId": 1234}});
  });
});

> //Handle a raised intent and return a Private Channel over which response will be sent

Author: ```suggestion
//Handle a raised intent and return a PrivateChannel over which response will be sent
```

getOrCreateChannel(channelId: string): Promise<Channel>;
```

> Returns a Channel object for the specified channel, creating it (as an _App_ channel) - if it does not exist.

Author: ```suggestion
Returns a `Channel` object for the specified channel, creating it (as an _App_ channel) - if it does not exist.
```

catch (err){
  //app could not register the channel
}
```

#### See also
*  [`Channel`](Channel)

### `createPrivateChannel`

```ts
createPrivateChannel(): Promise<PrivateChannel>;
```
 

> Returns a `Channel` with an auto-generated identity that is intended for private communication between applications. Primarily used to create Channels that will be returned to other applications via an IntentResolution for a raised intent.

Author: ```suggestion
Returns a `Channel` with an auto-generated identity that is intended for private communication between applications. Primarily used to create channels that will be returned to other applications via an IntentResolution for a raised intent.
```

If the `PrivateChannel` cannot be created, the returned promise MUST be rejected with an error string from the [`ChannelError`](Errors#channelerror) enumeration.

> The `PrivateChannel` type is provided to support synchronisation of data transmitted over returned channels, by allowing both parties to listen for events denoting subscription and unsubscription from the returned channel. `PrivateChannels` are only retrievable via raising an intent.

Author: `Channel` represents the Typescript object (and should be capitalized, as the type is, and code formatted), while channel represents the abstract concept from the spec. I've adjusted one instance on line 419 based on that.

It is intended that Desktop Agent implementations:
- SHOULD restrict external apps from listening or publishing on this channel.

> - MUST prevent private channels from being retrieved via fdc3.getOrCreateChannel.

Author: ```suggestion
- MUST prevent `PrivateChannels` from being retrieved via fdc3.getOrCreateChannel.
```

> - MUST provide the `id` value for the channel as required by the Channel interface.

Author: ```suggestion
- MUST provide the `id` value for the channel as required by the `Channel` interface.
```

#### Example

```js
fdc3.addIntentListener("QuoteStream", async (context) => {
	const channel = await fdc3.createPrivateChannel();
	const symbol = context.id.ticker;

	// This gets called when the remote side adds a context listener
	const addContextListener = channel.onAddContextListener((contextType) => {
		// broadcast price quotes as they come in from our quote feed
		feed.onQuote(symbol, (price) => {
			channel.broadcast({ type: "price", price});
		});
	});

	// This gets called when the remote side calls Listener.unsubscribe()
	const unsubscriberListener = channel.onUnsubscribe((contextType) => {
		feed.stop(symbol);
	});

	// This gets called if the remote side closes
	const disconnectListener = channel.onDisconnect(() => {
		feed.stop(symbol);
	});

	return channel;
});
```

#### See also

> *  [`PrivateChannel`](PrivateChannel)

Author: yep

//Raise an intent without a context by using the null context type
await fdc3.raiseIntent("StartChat", {type: "fdc3.nothing"});

//Raise an intent and retrieve a result from the IntentResolution
let resolution = await agent.raiseIntent("intentName", context);
try {
  const result = await resolution.getResult();
  if (result && result.broadcast) { //detect whether the result is Context or a Channel
    console.log(`${resolution.source} returned a channel with id ${result.id}`);

>   } else if (){

Author: yep
```suggestion
  } else if (result){
```

>   } else if (){

Author: ```suggestion
  } else if (result){
```
